### ISSUE TITLE
GPL Licensing

### ISSUE THREAD

**Issue by GijsTimmers** [GijsTimmers/kotnetcli#98]
Please update the license. We spoke about this a couple of weeks ago.

- [ ] `README.md`
- [ ] file headers
- [ ] ...

**Comment by jovanbulck** [comment]
Ok, first some notes for future reference:

- Currently `kotnetcli` is licensed under [CC-BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/), which is a free *as in freedom* license. 
- CC-BY-SA4 is [one-way compatible](https://creativecommons.org/compatiblelicenses/) with GPLv3. This means a contributor can opt to license his/her changes to a CC-BY-SA4 project under GPLv3 (but only these changes, and not the other way around).
- [GNU](http://www.gnu.org/philosophy/license-list.html#ccbysa) as well as [CC](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software) explicitly **discourage the use of CC licenses for software**. As these licenses are designed for artwork,  they do not include specifics on i.a. source code distribution, and may hinder integration in other free software projects.

I therefore propose to:

- Re-license the whole project (`master` as well as `dev` branch) "under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version."
- When relicensing, we need permission from all the copyright holders/contributors, which would be the 2 of us (apart from a marginal < 10 lines of code, which we can safely ignore imo).
- The way I see it, all of the existing code will be *dual licensed* CC-BY-SA4/GPLv3 and anyone can do what is allowed by any one of those licenses -- which (apart from small specifics) wont make a grand philosophical difference given the one-way compatibility described above. All future code from the relicensing on however will be exclusively licensed GPLv3.

I'll prepare a pull request for the above. Of course, the final call is @GijsTimmers .

PS: @samgielis we better explicitly license the artwork (logos, banner, etc)  separately under a (CC-BY-SA?) license? This is your call of course.

**Comment by samgielis** [comment]
fine by me! just use the license type that best fits your needs.

**Comment by GijsTimmers** [comment]
Sounds good - but how are we going to indicate what code is dual licensed and what code is GPLv3 licensed?

**Comment by jovanbulck** [comment]
> Sounds good - but how are we going to indicate what code is dual licensed and what code is GPLv3 licensed?

The way I see it, the entire existing code base is licensed CC-BY-SA4, and will continue to be so for ever and a day. We are the copyright holder of the code base however, and we can thus choose to **re-license** all existing code under GPLv3, simply by pushing the relevant commits in the `master` and `dev` branches. In this commit, we add the GPL license notice and remove the CC-BY-SA notive in all source code files. This implies that all code prior to that commit, is *dual licensed* CC-BY-SA4/GPLv3, and all future code changes from that commit on are licensed GPLv3-only.

In short, we do not have to indicate anything special. Normal future users will checkout the latest commit, and automatically be granted all GPLv3 freedoms. Should anyone, for any reason, want to do something that is only allowed by CC-BY-SA4, he/she should checkout an older version and not use any of the changes we made ever since the re-license commit.

I'll open a pull request with the proposed commits right now.

**Comment by jovanbulck** [comment]
This one is history now...
